nbdime/webapp/nbdifftool.py

```diff
@@ -59,11 +61,12 @@ def main(args=None):
     cwd = arguments.workdirectory
     base = arguments.base
     remote = arguments.remote
+    browsername = arguments.browser
     return run_server(
         port=port, cwd=cwd,
         closable=True,
         difftool_args=dict(base=base, remote=remote),
-        on_port=browse)
+        on_port=lambda port: browse(port, browsername))
```

Comment:
`on_port` is a callback for when the port has been established, so calling browse directly here will not work with random port.